agent/consul/acl_endpoint.go

```diff
@@ -458,8 +461,13 @@ func (a *ACL) TokenClone(args *structs.ACLTokenSetRequest, reply *structs.ACLTok
 	_, token, err := a.srv.fsm.State().ACLTokenGetByAccessor(nil, args.ACLToken.AccessorID, &args.ACLToken.EnterpriseMeta)
 	if err != nil {
 		return err
-	} else if token == nil || token.IsExpired(time.Now()) {
-		return acl.ErrNotFound
+	} else if token == nil {
+		if ns := args.ACLToken.EnterpriseMeta.NamespaceOrEmpty(); ns != "" {
+			return fmt.Errorf("token not found in namespace %s: %w", ns, acl.ErrNotFound)
+		}
+		return fmt.Errorf("token does not exist: %w", acl.ErrNotFound)
```

Comment:
minor / non-blocking: We do a sort of similar thing in multiple places. Is there a reasonable way to extract this into a re-usable function?

Maybe something simple like this.

```go
func aclResourceNotExistError(resourceType string, entMeta acl.EnterpriseMeta) error {
	if ns := entMeta.NamespaceOrEmpty(); ns != "" {
		return fmt.Errorf("%s not found in namespace %s: %w", resourceType, ns, acl.ErrNotFound)
	}
	return fmt.Errorf("%s does not exist: %w", resourceType, acl.ErrNotFound)
}

...
// And then call it like
return aclResourceNotExistError("token", args.ACLToken.EnterpriseMeta)
```

Reply:
done